Rethink vertical window resizing directions

janihrens opened the pull request:
The vertical vim-like resize bindings were swapped in comparison to the arrow keys.

j = Down = grow
k = Up = shrink

Commit:
Rethink vertical window resizing directions

The direction that windows are resized in are not visually consistent.
Top windows will expand down, bottom windows expand up and middle
windows expand in both directions.

Therefore thinking about resizing merely in terms of increasing and
reducing size of a window inspires the following mnemonics:
`growing Up` and `shrinking Down`.

janihrens commented:
Hey, this is just a quick edit in the web editor, so I haven't tested it. There shouldn't be any other moving parts involved though I think.

kgilmer commented:
Hi @mnkyhd thanks for your PR.  It is inconsistent between vim and arrows, you are correct. In looking at this in detail, I think it's more intuitive if "up" equates to bigger and "down" equates to smaller, and so the arrow keybindings are the ones that should change, rather than the vi bindings.  What do you think?

janihrens commented:
Hi @kgilmer, I've adapted the PR with your suggestion and added the reasoning to the commit message.

The use case where this came up for me was resizing a floating window that was beyond the edge of the screen when I switched from an attached monitor to the laptops display. The direction that floating windows grow is downwards. So the default seemed reasonable.

However when resizing tiling windows that are above each other this no longer applies. Your suggestion is an abstraction from this thinking that makes total sense IMO.
